core/templates/dev/head/domain/topic_viewer/TopicDataObjectFactory.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of frontend topic-data domain
+ * objects.
+ */
+
+export class TopicData {
```

Comment:
I would suggest renaming this to `RepresentableTopic` or `ViewableTopic` or `DisplayableTopic` or something around that. 

In general don't end the domain object name with `Data` as data is very generic and it doesn't explains what the domain object contain, like in this case what part of topic data it contains.

@aks681, do you have any suggestion on this?

Reply:
I think `ViewableTopic` sounds good, though can replace if there's better choice.

Reply:
This is for the viewer page, right? I would suggest `ReadOnlyTopic`, going by the other objects that are named similarly (ReadOnlyStoryNode and ReadOnlySubtopicPage).

Reply:
Using `ReadOnlyTopic` instead of `TopicData`